frontend/src/components/spatialChat/chatSettings/ChatSettings.tsx

```diff
@@ -0,0 +1,177 @@
+import {
+  Button,
+  FormControl,
+  FormLabel,
+  Input,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Select,
+  useDisclosure,
+  useToast,
+} from '@chakra-ui/react';
+import { MenuItem, Typography } from '@material-ui/core';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { ChatEditorType } from '../../../classes/SpatialChat';
+import {
+  changeBroadcastRadius,
+  changeEditorTypeAction,
+} from '../../../redux/actions/chatReducerActions';
+import { RootState } from '../../../redux/store';
+
+/**
+ * Component to update chat settings such as chat radius, chat editor, etc.
+ * @returns Chat Settings Component
+ */
+export default function ChatSettings() {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const chateditorOptions = [
+    { type: ChatEditorType.DEFAULT_EDITOR, text: 'Default Editor' },
+    { type: ChatEditorType.RICH_TEXT_EDITOR, text: 'Rich Text Editor' },
+  ];
+  const dispatch = useDispatch();
+  const currentEditorType: ChatEditorType = useSelector(
+    (state: RootState) => state.chat.settingChatEditorType,
+  );
+  const [lastEditorType, setLastEditorType] = useState<string | null>();
+  const currentRadius: number = useSelector(
+    (state: RootState) => state.chat.settingChatBroadcastRadius,
+  );
+  const [lastRadius, setLastRadius] = useState<string>();
+
+  const changeChatEditor = (editorType: ChatEditorType) => {
+    dispatch(changeEditorTypeAction(editorType));
+  };
+  const changeChatRadius = (radius: number) => {
+    dispatch(changeBroadcastRadius(radius));
+  };
+  useEffect(() => {
+    setLastRadius(currentRadius.toString());
+  }, [currentRadius]);
+
+  const openSettings = () => {
+    onOpen();
+  };
+
+  // Resetting the local states on close modal.
+  const closeSettings = () => {
+    setLastEditorType(null);
+    setLastRadius(currentRadius.toString());
+    onClose();
+  };
+
+  const toast = useToast();
+
+  const isNumeric = (str: any): boolean =>  !Number.isNaN(Number(str));
+
+  const validateSettings = () => {
+    if (!lastRadius || lastRadius?.trim().length === 0 || !isNumeric(lastRadius) ||  Number(lastRadius) < 80 || Number(lastRadius) > 1000)
+     {
+      toast({
+        title: 'Unable to update chat settings',
+        description: 'Broadcast Radius should be between 80 and 1000',
+        status: 'error',
+      });
+      return false;
+    }
+    return true;
+  };
+
+  const updateChatEditor = () => {
+    if (lastEditorType === '1') {
```

Comment:
**BUG** So this is going to create a problem. Here's a scenario where there is a bug because of this line - 
- Open chat-settings. (Observe the editor-type that you see on the screen)
- Change border-radius (Don't touch editor-type setting)
- Click submit 
You'll notice that the editor changes from default to rich-text. 

**Suggestion** 
This is all happening because lastEditorType here is null.
Similar to how border-radius was refactored, add a useEffect so that lastEditorType takes in the value in the redux-store initially. 

And then just re-write this to - 
```
 const updateChatEditor =  () => changeChatEditor(Number(lastEditorType))
```

You don't need to explicitly check what type it is, just passing in the number should be good enough. This is also related to the other comment I raised about setting hard-coded values to enums.